range_sensor_layer/src/range_sensor_layer.cpp

```diff
@@ -354,6 +358,23 @@ void RangeSensorLayer::updateCostmap(sensor_msgs::Range& range_message, bool cle
 
   buffered_readings_++;
   last_reading_time_ = ros::Time::now();
+  if(use_decay_)
+    timeCheck();
+}
+
+void RangeSensorLayer::timeCheck()
+{
+  std::map<std::pair<int,int>, double>::iterator it_map;
+  double last_reading_time_sec = last_reading_time_.toSec();
+
+  for (it_map = marked_point_history_.begin() ; it_map != marked_point_history_.end() ; it_map++ )
+  {
+    if(pixel_decay_ < last_reading_time_sec - it_map->second)
```

Comment:
If we really wanted to optimize, we could calculate `removal_time = last_reading_time.toSec() - pixel_decay_` and then simply check if `it_map->second < removal_time`